### ISSUE TITLE
Why sometimes it returns the same neighbor twice

### ISSUE THREAD

**Issue by zyxue** [storpipfugl/pykdtree#44]
![image](https://user-images.githubusercontent.com/527862/57493565-550cf580-727a-11e9-96dd-6ac392e88645.png)

As seen in the first row, 460448, 441197, 83629, 208359 are returned twice, what could be the reason, please?

**Comment by zyxue** [comment]
This problem becomes much more serious when `k=100`

**Comment by djhoese** [comment]
Just curious and just a guess, but are your input lng/lat points unique? I wonder if some of them are repeated and that's confusing the kdtree.

**Comment by zyxue** [comment]
yes, they're unique. I switched to [scipy.spatial.KDTree](https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.spatial.KDTree.html#scipy-spatial-kdtree), and not seeing the same problem.

**Comment by mraspaud** [comment]
@storpipfugl any idea ?

**Comment by storpipfugl** [comment]
pykdtree has been validated against scipy.spatial.KDTree so disagreement of results is unexpected. @zyxue : Is it possible for you to provide a minimal data set that reproduces the problem?
